File: hosung/programmers/문자열 압축.py
def solution(s):

    answer = 2147000000
    for j in range(1, len(s)+1):
        res = ""
        cnt = 1
        ep = s[:j]

        for i in range(j,len(s),j):
            if s[i:i+j] == ep:
                cnt+=1
            else:
                if cnt>1:
                    res += str(cnt) + ep
                else:
                    res+=ep
                cnt = 1
                ep = s[i:i+j]
        if cnt >1:
            res += str(cnt) +ep
        else:
            res+=ep
        if len(res) < answer:
            answer = len(res)


    return answer
            
        
                
                
                

File: taejoong/programmers/문자열압축.java
class Solution {
    public int solution(String s) {
        int answer = s.length();
        
        for(int i = 1; i <= s.length() / 2; i++) {
            StringBuilder sb = new StringBuilder();
            String prev = s.substring(0,i);
            int cnt = 1;
            
            for(int j = i; j < s.length(); j+=i) {
                int end = Math.min(j+i, s.length());
                String cur = s.substring(j, end);
                
                if(prev.equals(cur)) {
                    cnt++;
                } else {
                    if(cnt > 1) {
                        sb.append(cnt);
                    }
                    sb.append(prev);
                    cnt = 1; 
                    prev = cur;
                }               
                                              
            }
            
            if (cnt > 1) {
                sb.append(cnt);
            }
            sb.append(prev);
            answer = Math.min(answer, sb.length());
        }
        return answer;
    }                
}